app.py

```diff
@@ -0,0 +1,244 @@
+            visible, empty string if visible.
+
+    Returns:
+        str: The new class name of the thing to collapse.
+    """
+
+    classes = to_collapse_class.split(" ") if to_collapse_class else []
+    if "collapsed" in classes:
+        classes.remove("collapsed")
+        return " ".join(classes)
+    return to_collapse_class + " collapsed" if to_collapse_class else "collapsed"
+
+
+@app.callback(
+    Output("input", "children"),
+    inputs=[
+        Input("slider", "value"),
+    ],
+)
+def render_initial_state(slider_value: int) -> str:
+    """Runs on load and any time the value of the slider is updated.
+        Add `prevent_initial_call=True` to skip on load runs.
+
+    Args:
+        slider_value: The value of the slider.
+
+    Returns:
+        str: The content of the input tab.
+    """
+    return f"Put demo input here. The current slider value is {slider_value}."
+
+
+class RunOptimizationReturn(NamedTuple):
+    """Return type for the ``run_optimization`` callback function."""
+
+    results: str = dash.no_update
+    problem_details_table: list = dash.no_update
+    # Add more return variables here. Return values for callback functions
+    # with many variables should be returned as a NamedTuple for clarity.
+
+
+@app.callback(
+    # The Outputs below must align with `RunOptimizationReturn`.
+    Output("results", "children"),
+    Output("problem-details", "children"),
+    background=True,
+    inputs=[
+        Input("run-button", "n_clicks"),
+        State("sampler-type-select", "value"),
+        State("solver-time-limit", "value"),
+        State("slider", "value"),
+        State("dropdown", "value"),
+        State("checklist", "value"),
+        State("radio", "value"),
+    ],
+    running=[
+        # Shows cancel button while running.
+        (Output("cancel-button", "className"), "", "display-none"),
+        (Output("run-button", "className"), "display-none", ""),  # Hides run button while running.
+        (Output("results-tab", "disabled"), True, False),  # Disables results tab while running.
+        (Output("results-tab", "label"), "Loading...", "Results"),
+        (Output("tabs", "value"), "input-tab", "input-tab"),  # Switch to input tab while running.
+        (Output("run-in-progress", "data"), True, False),  # Can block certain callbacks.
+    ],
+    cancel=[Input("cancel-button", "n_clicks")],
+    prevent_initial_call=True,
+)
+def run_optimization(
+    # The parameters below must match the `Input` and `State` variables found
+    # in the `inputs` list above.
+    run_click: int,
+    sampler_type: Union[SamplerType, int],
+    time_limit: float,
+    slider_value: int,
+    dropdown_value: int,
+    checklist_value: list,
+    radio_value: int,
+) -> RunOptimizationReturn:
+    """Runs the optimization and updates UI accordingly.
+
+    This is the main function which is called when the `Run Optimization` button is clicked.
```

Comment:
Not important here since no docs are built (double backticks are used for our doc builds with Sphinx), but just for consistency.
```suggestion
    This is the main function which is called when the ``Run Optimization`` button is clicked.
```